### PR TITLE
docs - new sql ref pages for CREATE/DROP ACCESS METHOD

### PR DESCRIPTION
**Pull request by lisakowen**
add sql reference pages for CREATE ACCESS METHOD and DROP ACCESS METHOD.  i pulled the postgres 12 content.

question:  is there any greenplum-specific info that should be added to these pages?

doc review site link (behind vpn):  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/sqlref-cam/greenplum-database/GUID-ref_guide-sql_commands-CREATE_ACCESS_METHOD.html

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Only superusers can define new access methods.
+
+
+## <a id="section4"></a>Parameters 
+
+name
+:   The name of the access method to create.
+
+TYPE access\_method\_type
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md:27]: Remove "TYPE" here? (to conform with the postgres docs and the "Parameters" topic type?

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+:   The type of access method to define. Only `TABLE` and `INDEX` types are supported at present.
+
+HANDLER handler\_function
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md:30]: Same here - remove "HANDLER".

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+:   handler\_function is the name \(possibly schema-qualified\) of a previously registered function that represents the access method. The handler function must be declared to take a single argument of type `internal`, and its return type depends on the type of access method; for `TABLE` access methods, it must be `table_am_handler` and for `INDEX` access methods, it must be `index_am_handler`. The C-level API that the handler function must implement varies depending on the type of access method. The table access method API is described in [Table Access Method Interface Definition](https://www.postgresql.org/docs/12/tableam.html) in the PostgreSQL documentation. The index access method API is described in [Index Access Method Interface Definition](https://www.postgresql.org/docs/12/indexam.html).
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md:31]: handler\_function -> `handler_function`

Comment (dyozie) [reply gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md:31]: Also - no need to escape the parens in the paragraph body.

File: gpdb-doc/markdown/ref_guide/sql_commands/DROP_ACCESS_METHOD.html.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+## <a id="section4"></a>Parameters 
+
+IF EXISTS
+:   Do not throw an error if the access method does not exist. Greenplum Database issues a notice in this case.
+
+name
+:   The name of an existing access method.
+
+CASCADE
+:   Automatically drop objects that depend on the the access method \(such as operator classes, operator families, and indexes\), and in turn all objects that depend on those objects.
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/sql_commands/DROP_ACCESS_METHOD.html.md:24]: repeated word:  the the

File: gpdb-doc/markdown/ref_guide/sql_commands/DROP_ACCESS_METHOD.html.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+RESTRICT
+:   Refuse to drop the access method if any objects depend on it. This is the default.
+
+## <a id="section5"></a>Examples 
+
+Drop the access method heptree;
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/sql_commands/DROP_ACCESS_METHOD.html.md:31]: heptree -> `heptree`

**Review by dyozie — CHANGES_REQUESTED** [review]

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md (hunk 3 above)
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/sql_commands/CREATE_ACCESS_METHOD.html.md:31]: thanks for reviewing.  a few formatting-related questions:
- i am not sure why we would code-font handler_function in paragraph text but not in the definition list?  most of the sql reference pages do not code font (or highlight in any other manner) the parameters that were previously `<xxx>` or italicized in the synopsis.  (i think this originated with the dita to markdown conversion.)
- why do we not need to escape the parens in the paragraph body?  when do we need to escape parens?

**Review by AJR-VMware — COMMENTED** [review]
# Review done
* create access method AJR type table handler heap_tableam_handler;
    * worked
* create access method AJR2 type index handler bthandler;
    * worked
* create access method AJR3 type index handler heap_tableam_handler;
    * failed as expected
* create access method AJR3 type table handler brinhandler;
    * failed as expected
* create access method AJR3 type DNE handler heap_tableam_handler;
    * failed as expected
* drop access method ajr;
    * worked
* drop access method ajr2;
    * worked
* drop access method ajr3;
    * failed as expected
* many tests exist in src/test for both create and drop AM
    * regress/sql/create_am.sql
    * modules/test_pg_dump/sql/test_pg_dump.sql
    * regress/sql/drop_if_exists.sql
    * regress/sql/psql.sql

# Findings
* Compatibility section -- CREATE ACCESS METHOD is a Greenplum Database extension.
    * This is present verbatim in current upstream.  Do we have separate language to indicate that?

**Comment by AJR-VMware** [comment]
@lisakowen I think you're right, there's no changes needed here.  I see the pattern we're using, upstream calls it out as an extension in the same way.

Thanks for responding, looks good to me.

**Review by dyozie — APPROVED** [review]

**Pull request merged**